### PR TITLE
WIP: fix mypy issues

### PR DESCRIPTION
**Pull request by arnaudgelas**
**Features**

    
**Feature Docs**

**Influence**

**Result**

**Other**

### CODE REVIEW

**Comment by arnaudgelas** [comment]
@geekan This kind of PR requires coordination, but also the ability to run unit tests

**Comment by geekan** [comment]
It's nice to see contributing to the code. Since we now have a more dev branch than main, can you commit on the dev branch?

File: metagpt/config.py [hunk 1]
```diff
@@ -125,6 +125,7 @@ def get_default_llm_provider_enum(self) -> LLMProviderEnum:
         if provider:
             logger.info(f"API: {provider}")
             return provider
+        return LLMProviderEnum.OPENAI
```
Comment (better629) [inline metagpt/config.py]: no need add this, previous code has add DEFAULT_PROVIDER.
but it seems missing DEFAULT_PROVIDER here ==

File: metagpt/document.py [hunk 2]
```diff
@@ -118,23 +120,23 @@ class IndexableDocument(Document):
 
     model_config = ConfigDict(arbitrary_types_allowed=True)
 
-    data: Union[pd.DataFrame, list]
+    data: Union[pd.DataFrame, List[Any]]
```
Comment (better629) [inline metagpt/document.py:123]: use list, same as below

File: metagpt/environment.py [hunk 3]
```diff
@@ -41,16 +41,17 @@ def init_roles(self):
         self.add_roles(self.roles.values())
         return self
 
-    def serialize(self, stg_path: Path):
+    def serialize(self, stg_path: Path) -> None:
         roles_path = stg_path.joinpath("roles.json")
-        roles_info = []
+        roles_info: List[Dict[str, Any]] = []
         for role_key, role in self.roles.items():
+            r = self.members.get(role)
```
Comment (better629) [inline metagpt/environment.py]: we have a update in dev branch, maybe you can use dev.

File: metagpt/actions/action_node.py [hunk 4]
```diff
@@ -81,7 +81,7 @@ def __init__(
         instruction: str,
         example: Any,
         content: str = "",
-        children: dict[str, "ActionNode"] = None,
+        children: Optional[Dict[str, "ActionNode"]] = None,
```
Comment (better629) [inline metagpt/actions/action_node.py:150]: which tool do you used to find the increment content of the PR?

File: metagpt/environment.py (hunk 3 above)
Comment (arnaudgelas) [reply metagpt/environment.py]: I was not aware of the used branching model. As there is no documentation on the way you guys develop, nor how to contribute (or if there is any, can you make it clear in the readme, pls?)

File: metagpt/actions/action_node.py (hunk 4 above)
Comment (arnaudgelas) [reply metagpt/actions/action_node.py:150]: I have been using `mypy` and a bit of `pyright`. I started the thing cause I faced a couple of exceptions when trying to do a simple experiment. So I started with the files that were close to the issues.

File: metagpt/environment.py (hunk 3 above)
Comment (better629) [reply metagpt/environment.py]: currently, we update changes in dev and small release-fix in main. So you can merge the changes into dev. And we use pre-commit to check the style.  There still are some `List` `Dict` in the repo, and we will update to `list` `dict`, so just use the new ones.

**Pull request closed without merging**